osu.Game/Beatmaps/IO/BeatmapArchiveReader.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System;
+using osu.Framework.Logging;
+using osu.Framework.Platform;
+using osu.Game.Database;
+
+namespace osu.Game.Beatmaps.IO
+{
+    public abstract class BeatmapArchiveReader : ArchiveReader
+    {
+
+        public static BeatmapArchiveReader GetBeatmapArchiveReader(Storage storage, string path)
+        {
+            try
+            {
+                return (BeatmapArchiveReader)GetReader(storage, path);
+            }
+            catch (InvalidCastException e)
+            {
+                Logger.Error(e, "A tricky ArchiveReader instance passed the test to be a BeatmapArhiveReader, but it's really not");

```

Comment:
use `$@"{nameof(BeatmapArchiveReader)}"` to avoid typos (you have a typo in the name here).